Add dynamic name lengths

Snowstar38 opened the pull request:
- Cats' names in the family tree and cat list screens will now be limited by the actual width in pixels, rather than a set # of characters

Previously, if there was a cat named "llllllllll" and a cat named "mmmmmmmmmm", they would both be cut off at the same # of characters, despite one being much shorter visually.

Snowstar38 committed:
Add dynamic name lengths

- Cats' names in the family tree and cat list screens will now be limited by the actual width in pixels, rather than a set # of characters

Previously, if there was a cat named "llllllllll" and a cat named "mmmmmmmmmm", they would both be cut off at the same # of characters, despite one being much shorter visually.

Snowstar38 committed:
Apparently python hates ellipsis

Snowstar38 committed:
Dynamic name lengths for mediator screen

keyraven reviewed (changes requested):
I like this change! My only note is that there is a lot of duplicated code for handling the length of the string. It would be more tidy to create a new utility function to handle the length adjustments. This will also allow easy tweaking of this behavior.

Snowstar38 committed:
Add shorten_text_to_fit function

- Created function to shorten text based on length_limit, font_size, and (optionally) font_type
- Replaced every len(name) (but the ones in inheritance.py) to shorten_text_to_fit

Snowstar38 committed:
Merge branch 'Thlumyn:development' into Misc-changes

Snowstar38 committed:
Name placement tweaking

- This also fixes the issue with the favorite button being a variable distance from the name

keyraven commented on scripts/screens/cat_screens.py:
```diff
@@ -2357,10 +2356,11 @@ def update_selected_cat(self):
-                                                                             name,
+        length_limit = 300
+        font_size = 26
+        short_name = shorten_text_to_fit(name, length_limit, font_size)
```
I suggest removing the variables "length_limit" and "font_size", and simple pass these values directly into shorten_text_to_fix.  These variables are not used outside of the function call, and removing unnecessary variables increase readability.

keyraven reviewed (changes requested):
Looks pretty good! I just have one small note, but other than that, it's a very nice addition!

Snowstar38 committed:
Cleaning up shorten_text_to_fit
